adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -1707,3 +1731,33 @@ func addSubscriptionRatelimitActions(actions []*routev3.RateLimit) []*routev3.Ra
 			},
 		})
 }
+
+// Validate envoy configuration for smileys/special characters/escape characters. Including regex validation
+func validateEnvoyRoutePathString(configValue string) error {
+	var smileyValidationRegex string = "[:;=Xx8][-~]?[)DPOp3\\[\\]{}]|<3|:\\^\\)|:[)]|:[(]|:[/]|:[Pp]|:[Dd]|:[Oo]|:[|]|:S|:>|:[oO]"
+	smileyValidated, err := regexp.MatchString(smileyValidationRegex, configValue)
+	if err != nil {
+		return err
+	}
+	var substitutionStringValidationRegex string = `\{\{\s*[^}]+\s*\}\}|\$\{\s*[^}]+\s*\}|\{%\s*[^%]+\s*%\}|<%[-=]?\s*[^%]+?\s*%>|%[^%\s]+%`
+	subValidated, err := regexp.MatchString(substitutionStringValidationRegex, configValue)
+	if err != nil {
+		return err
+	}
+	if smileyValidated && subValidated {
+		return nil
+	} else if !smileyValidated {
+		return errors.New("route path contains an emoticon")
+	}
+	return errors.New("route path contains a substitution string")
+}
+
+// Validate envoy route method whether its of known type
+func validateEnvoyRouteMethodString(method string) error {
```

Comment:
From enforcer what we support is anyway "GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS". 
Shall we remove the rest

Reply:
Thanks, Fixed.